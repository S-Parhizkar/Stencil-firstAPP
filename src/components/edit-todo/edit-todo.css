:host {
  display: block;
}
.body{
  margin: 0 15px;
  height: auto;
}

.modal{
  display: block;
  background: rgb(60 221 239);
  padding: 10px;
  border-radius: 10px;
  overflow:hidden;
}
.label{
  color: #4d1403;
  font-size: 15px;
  
 
}
.inputPop{
  color: #c03e1e;
  border-radius: 3px;
  margin: 5px 0 5px 25px;
}

.confirm{
    box-shadow: inset 0px 1px 0px 0px #77eb94;
    background: linear-gradient(to bottom, #2bdb48 5%, #02640f 50%);
    background-color: #0aee3b;
    border-radius: 3px;
    border: 1px solid #013f1d;
    display: inline-block;
    cursor: pointer;
    color: #ffffff;
    font-family: Arial;
    font-size: 13px;
    padding: 9px 10px;
    text-decoration: none;
    text-shadow: 0px 1px 0px #854629;
    margin-left: 80%;

}
.confirm:hover {
  background: linear-gradient(to bottom, #066f3a 5%, #2acc2a 100%);
  background-color: #0aa74b;
}


h3 {
 margin: 0;
}
