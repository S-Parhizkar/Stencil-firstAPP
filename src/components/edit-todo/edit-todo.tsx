import { Component, h, Prop, Event, EventEmitter, Listen, State } from '@stencil/core';
import { ITodo } from '../types';

@Component({
  tag: 'edit-todo',
  styleUrl: 'edit-todo.css',
  shadow: true,
})
export class EditTodo {
  @Prop() todo: ITodo;
 
  private $textInputEditTitle?: HTMLInputElement;
  private $textInputEditOrder?: HTMLInputElement;
  @Event({ eventName: 'todo-to-edit' }) editTodoHandler: EventEmitter<ITodo>;

  handleEditTodo(todo: ITodo) {
    const title = this.$textInputEditTitle.value;
    const order = Number(this.$textInputEditOrder.value);
    const newTodo = {
      ...this.todo,
      title,
      order,
    };
    this.editTodoHandler.emit(newTodo );
    console.log(todo)
    // this.emptyeditInputValue();
  }
  // Empty input
  // emptyeditInputValue() {
  //   this.$textInputEditTitle.value = '';
  //   this.$textInputEditOrder.value = '';
  // }


  render() {
    return (
      <div class= 'modal'>
        <h3>You can edit your 'to do's content from here: </h3>

        <hr />
        <div class='body'>
          <label class="label">
            <b>Replace your target 'to do's Title :</b>
          </label>
          <input
            class='inputPop'
            type="text"
            placeholder="can change TITLE "
            name="title"
            required
            value={this.todo?.title}
            ref={el => (this.$textInputEditTitle = el as HTMLInputElement)}
          />
          <br />
          <label class="label">
            <b>Replace your target 'to do's order :</b>
          </label>
          <input
            class='inputPop'
            type="number"
            placeholder="can change ORDER"
            name="order"
            required
            value={this.todo?.order}
            ref={el => (this.$textInputEditOrder = el as HTMLInputElement)}
          />
          <h5> ID : {this.todo.id}</h5>

          <button type="submit" class='confirm' onClick={() => this.handleEditTodo(this.todo)}>
            Confirm
          </button>
        </div>
      </div>
    );
  }
}
