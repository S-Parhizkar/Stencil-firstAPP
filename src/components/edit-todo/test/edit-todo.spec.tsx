import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { EditTodo } from '../edit-todo';

describe('edit-todo', () => {
  const todo = {
    completed: true,
    id: 2,
    order: 3,
    title: 'sport',
    url: 'https://my-list-to-do.netlify.app/',
  };

  let page: SpecPage

  beforeEach(async () => {
    page = await newSpecPage({
      components: [EditTodo],
      template: () => <edit-todo todo={todo}></edit-todo>,
    });
    await page.waitForChanges();
  });

  it('render', () => {
    expect(page.root).toMatchSnapshot();
  });

  it('should emit [todo-to-edit] event when the todo updated is completed', async () => {
    const listenerMock = jest.fn();
    page.body.addEventListener('todo-to-edit', listenerMock);
    expect(listenerMock).not.toHaveBeenCalledWith();
    (page.root.shadowRoot.querySelector('button.confirm') as HTMLButtonElement).click();
    await page.waitForChanges();
    const mockTodo = { ...todo };
    mockTodo.title = todo.title.toUpperCase().trim();
    expect(listenerMock).toHaveBeenCalledWith(expect.objectContaining({ detail: mockTodo}));

  });

  it('should emit [todo-to-edit] event with updated todo when the todo updated is completed', async () => {
    const listenerMock = jest.fn();
    page.body.addEventListener('todo-to-edit', listenerMock);

    (page.root.shadowRoot.querySelector('input.input-order') as HTMLInputElement).value = 100 as  unknown as string;
    (page.root.shadowRoot.querySelector('button.confirm') as HTMLButtonElement).click();
    await page.waitForChanges();
    const mockTodo = { ...todo };
    mockTodo.title = todo.title.toUpperCase().trim();
    mockTodo.order = 100;
    expect(listenerMock).toHaveBeenCalledWith(expect.objectContaining({ detail: mockTodo}));

  });
});
