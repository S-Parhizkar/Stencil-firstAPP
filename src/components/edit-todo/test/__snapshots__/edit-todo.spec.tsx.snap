// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit-todo render 1`] = `
<edit-todo>
  <mock:shadow-root>
    <div class="modal">
      <h3>
        You can edit your 'to do's content from here:
      </h3>
      <hr>
      <div class="body">
        <label class="label">
          <b>
            Replace your target 'to do's Title :
          </b>
        </label>
        <input class="input-title" name="title" placeholder="can change TITLE " required="" type="text" value="sport">
        <br>
        <label class="label">
          <b>
            Replace your target 'to do's order :
          </b>
        </label>
        <input class="input-order" name="order" placeholder="can change ORDER" required="" type="number" value="3">
        <h5>
          <span class="label">
            ID :
          </span>
          <span>
            2
          </span>
        </h5>
        <hr>
        <div class="container">
          <button class="cancel" type="submit">
            Cancel
          </button>
          <button class="confirm" type="submit">
            Confirm
          </button>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</edit-todo>
`;
